libs/sdk-ui-dashboard/src/presentation/topBar/buttonBar/button/selectors.ts

```diff
@@ -104,25 +75,28 @@ export const selectIsCurrentDashboardVisibleInList = createSelector(
  * @internal
  */
 export const selectIsShareButtonVisible = createSelector(
-    selectCanManageWorkspace,
-    selectHasPermissionsForShare,
-    selectIsCurrentDashboardVisibleInList,
+    selectEnableAnalyticalDashboardPermissions,
+    selectCanShareDashboardPermission,
+    selectCanShareLockedDashboardPermission,
     selectDashboardLockStatus,
+    selectIsCurrentDashboardVisibleInList,
     selectIsReadOnly,
     selectIsInEditMode,
     selectIsShareButtonHidden,
     (
-        isAdmin,
-        hasPermission,
-        isCurrentDashboardVisibleInList,
+        dashboardPermissionsEnabled,
+        canShareDashboardPermission,
+        canShareLockedDashboardPermission,
         isLocked,
+        isCurrentDashboardVisibleInList,
         isReadOnly,
         isInEditMode,
         isShareButtonHidden,
     ) =>
-        hasPermission &&
+        dashboardPermissionsEnabled &&
+        canShareDashboardPermission &&
+        (!isLocked || canShareLockedDashboardPermission) &&
```

Comment:
In this function, again `canManageWorkspace` was replaced with canShareLockedDashboardPermission (which is based on SHARE dashboard permission on Tiger and `canManageWorkspace` WS permission on Bear)